_posts/2018-03-31-rereading-the-mythical-man-month.md

```diff
@@ -0,0 +1,207 @@
+
+In **1971**, **Niklaus Wirth** formalized a design procedure by a sequence of **refinement steps**:
+
+- _One sketches a rough task definition and a rough solution method that achieves the principal result._
+- _One examines the definition more closely to see how the result differs from what is wanted._
+- _One takes the large steps of the solution and breaks them down into smaller steps._
+
+_From this process one identifies modules of solution or of data whose further refinement can proceed independently of other work._
+
+_The degree of this **modularity** determines the adaptability and changeability of the program._
+
+# Communication in source code
+
+It's said that _a computer program is a message from a man to a machine_. On the other hand, _it tells its story to the human user_, so documentation was important. In order to solve the problem with the maintenance of that documentation, they decided to write **self-documenting programs**:
+
+* **Using the parts of the program** to carry as much of the documentation as possible to convey as much **meaning** as possible to the reader:
+    * Labels
+    * Declaration statements
+    * Symbolic names
+* Using space and format as much as possible to improve **readability** and show subordination and nesting
+* Inserting comments, not too many
+
+The drawback of such an approach to documentation was the increase in the size of the source code that might be stored.
+
+# No Silver Bullet
+
+Brooks wrote that there was no silver bullet:
+
+> There is no single development, in either technology or management technique, which by itself promises even one order-of-magnitude improvement within a decade in productivity, in reliability, in simplicity.
+
+Following Aristotle, he talked about two types of difficulties in software projects:
+
+* **Essence**: inherent in the nature of the software.
+    * Complexity with a large number of states
+    * Conformity to other interfaces
+    * Changeability for new cases
+    * Invisibility, not inherently embedded in space and we need several graphs to represent the software
+* **Accidents**: incidental complexities that are imposed by the ways that software is built in the world.
+    * Physical limitations of size and speed
+    * Low levels of abstraction in programming languages
+    * Imperfect specification or interpretation of requirements
+    * Imperfect and inefficient communication among developers
+    * ...
+
+Brooks prized technical developments as _potential silver bullets_: high-level languages advances, OOP, Artificial Intelligence, Expert Systems, _automatic_ programming, graphical programming, program verification (not _error-proof programs_, but _meeting specification_), environments, tools and workstations. On the other hand, he valued _buy vs. build_, requirements refinement and rapid prototyping, incremental development (growing, not building, software) and being great designers, because _software construction is a creative process_.
```

Comment:
Prized? Do you mean appraised maybe? I read it recently too (you inspired me), and unless my memory is faulty, Brooks held his original assertion as still true, that there is still no silver bullet.

Reply:
Yes, there is still no silver bullet, but he considers there are "potencial" silver bullets.

Reply:
Okay, so "prized" means "considers extremely valuable," is that your intended meaning?

Reply:
Yes! Thanks @richardjwild !!